client/windows/CCreatureWindow.cpp

```diff
@@ -324,6 +324,11 @@ CStackWindow::ButtonsSection::ButtonsSection(CStackWindow * owner, int yOffset)
 
 			upgradeBtn->addOverlay(std::make_shared<CAnimImage>("CPRSMALL", VLC->creh->objects[upgradeInfo.info.newID[buttonIndex]]->iconIndex));
 
+			if(buttonIndex == 0)

```

Comment:
any chance to get rid of the magic value?

Reply:
What do you suggest? FIRST_BUTTON_INDEX = 0? LOWEST_UPGRADE_INDEX = 0?

Reply:
if you give names to all indexes (e.g. constants), then you can iterate through them as an initialiser list (for-in) instead of plain for. Or have an enum with last element named Count or so.

Reply:
The goal of constant here is to give some hint on what the code actually does (it is onetime usage). Why 0 is there. Have you understood why? If yes - which constant reflects it the best? If it does not help understanding - it is useless here and plain number might be even better

Reply:
Also there is some danger in this solution that both upgrades might be equivalent and you just can not automatically decide which one to hit when there is more than one. So we can do it another way and allow shortcut only if there is single upgrade available

Reply:
thanks, now it's clear that there can be multiple upgrades, and 0 simply means "first".